### PR TITLE
[ENH] Network refactor

### PR DESCRIPTION
**Pull request by rksh**
Re-opened due to large change while in-flight.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 1]
```diff
@@ -2405,6 +2410,27 @@ public void endRouteMapping(SharedPreferences prefs) {
...
+     * When prefs/the token manager have been updated in response to authorization/de-auth, recreate in order to reflect the state change.
+     */
+    public static void refreshApiManager() {
+        final MainActivity mainActivity = MainActivity.getMainActivity();
+        if (null != mainActivity) {
+            MainActivity.State s = mainActivity.getState();
+            if (null != s) {
+                SharedPreferences prefs = mainActivity.getPreferences(MODE_PRIVATE);
+                if (null != prefs) {
+                    mainActivity.state.apiManager = new WiGLEApiManager(prefs, mainActivity.getApplicationContext());
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java:2423]: this ... seems odd? should this be s and not the member access?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/RankStatsFragment.java [hunk 2]
```diff
@@ -171,55 +141,69 @@ private void setupSwipeRefresh(final LinearLayout rootView) {
...
+            }
+            final boolean doMonthRanking = monthRanking.get();
+            final String sort = doMonthRanking ? "monthcount" : "discovered";
+            long pageStart = 0;
+            long pageEnd = ROW_COUNT;
+            long selected = 0;
+            Long userRank;
+            if (userCentric.get()) {
+                final String userRankKey = doMonthRanking ? ListFragment.PREF_MONTH_RANK : ListFragment.PREF_RANK;
+                final SharedPreferences prefs = getActivity().getSharedPreferences(PreferenceKeys.SHARED_PREFS, 0);
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/RankStatsFragment.java:164]: is this getActivity() different from fragmentActivity above?

**Review by uhtu — APPROVED** [review]
that's alotta debugged code 👍

**Comment by bobzilladev** [comment]
Are any of your changes different from last time, or is this just a rebase type thing?

**Review by uhtu — APPROVED** [review]

**Comment by rksh** [comment]
Adds refactor of DBResultAcitivty (and hence a new Network constructor)

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java (hunk 1 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: quite right

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+            @Override
+            public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                onCallFailure("Unsuccessful WiGLE Uploads request: ", e,
+                        completedListener, mainHandler, null);
+            }
+        });
+    }
+
+    public void searchWiFi(@NotNull final String urlEndodedQueryParams,
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:369]: "urlEndodedQueryParams" these params are well endoded!

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/RankStatsFragment.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/RankStatsFragment.java]: pure brain fart / caution for being in a static method.

**Review by bobzilladev — APPROVED** [review]

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DBResultActivity.java [hunk 4]
```diff
@@ -291,29 +209,57 @@ public void complete() {
...
+                        resultList.clear();
+                    });
                 } else {
-                    handler.sendEmptyMessage(MSG_QUERY_EMPTY);
+                    handler.post(() -> handleFailedRequest());
                 }
             }
         }, ListFragment.lameStatic.dbHelper);
-
+Logging.info("===executing request...");
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DBResultActivity.java]: yep!

**Pull request merged**